File: src/cpp/sorts/SelectSort.cpp
#include<iostream>
#include<vector>

class SelectSort{
public:
    void selectSort(std::vector<int>& arr){
        int len = arr.size();
        if (len <= 1)
        {
            return;
        }
        for (int i = 0; i < len - 1; i++)
        {
            int minIndex = i;
            for (int j = i + 1; i < len; i++)
            {
                if (arr[j] < arr[minIndex])
                {
                    minIndex = j;
                }
            }
            int tmp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = tmp;
        }
        
    }
};

File: src/java/sorts/SelectSort.java
package java.sorts;

/**
 * @fileName: SelectSort.java
 * @author: wyli
 * @createTime: 2023/9/3 22:00
 * @description:
 */
public class SelectSort {

    public static void selectSort(int[] a, int n){
        if (n <= 1) return;
        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (a[j] < a[minIndex]){
                    minIndex = j;
                }
            }
            int tmp = a[i];
            a[i] = a[minIndex];
            a[minIndex] = tmp;
        }
    }

}
